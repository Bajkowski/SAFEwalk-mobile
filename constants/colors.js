const colors = {
  // red: "#b52f28",
  red: "#c5050c",
  darkred: "#9b0000",
  orange: "#ed7c3a",
  white: "#fff"
};

export default colors;
