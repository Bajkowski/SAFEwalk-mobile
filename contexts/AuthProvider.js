import React, { createContext, useReducer, useMemo } from "react";
import { AsyncStorage } from "react-native";

export const AuthContext = createContext();

export function AuthProvider({ children }) {
  const [state, dispatch] = useReducer(authReducer, {
    // initial states
    isLoading: true,
    isSignout: false,
    userToken: null,
    userType: "user",
    userEmail: null
  });

  // Handle Login
  async function login(userType, { email, password }) {
    try {
      // Put code below for this: make login GET request to server

      // For now, while server is not set, use email as value for user token
      await AsyncStorage.setItem("userToken", JSON.stringify(email));
      // Store the user type i.e. which type of user is logged in, user or SAFEwalker.
      await AsyncStorage.setItem("userType", userType);
      // Store the user email
      await AsyncStorage.setItem("userEmail", JSON.stringify(email));

      dispatch({
        type: "LOG_IN",
        token: JSON.stringify(email),
        userType: userType,
        userEmail: email
      });
    } catch (error) {
      throw new Error("Error in login(): " + error);
    }
  }

  // Handle signout
  async function signout() {
    try {
      await AsyncStorage.removeItem("userToken");
      await AsyncStorage.removeItem("userType");
      await AsyncStorage.removeItem("userEmail");

      dispatch({ type: "SIGN_OUT" });
    } catch (error) {
      throw new Error("Error in signout(): " + error);
    }
  }

  // Handle register
  async function register({
    email,
    password,
    firstName,
    lastName,
    phoneNumber
  }) {
    try {
      // Put code below for this: make register POST request to server

      // For now, while server is not set, use email as value for user token
      await AsyncStorage.setItem("userToken", JSON.stringify(email));
      // Store the user type i.e. which type of user is logged in, user or SAFEwalker.
      await AsyncStorage.setItem("userType", userType);
      // Store the user email
      await AsyncStorage.setItem("userEmail", JSON.stringify(email));

      dispatch({
        type: "LOG_IN",
        token: JSON.stringify(email),
        userType: "user",
        userEmail: email
      });
    } catch (error) {
      throw new Error("Error in register(): " + error);
    }
  }

  const authContextValue = useMemo(() => {
    return {
      isLoading: state.isLoading,
      isSignout: state.isSignout,
      userToken: state.userToken,
      userType: state.userType,
      userEmail: state.userEmail,
      dispatch,
      login,
      signout,
      register
    };
  }, [state]);

  return (
    <AuthContext.Provider value={authContextValue}>
      {children}
    </AuthContext.Provider>
  );
}

function authReducer(prevState, action) {
  switch (action.type) {
    case "RESTORE_TOKEN":
      return {
        ...prevState,
        isLoading: false,
        userToken: action.token,
        userType: action.userType,
        userEmail: action.userEmail
      };
    case "LOG_IN":
      return {
        ...prevState,
        isSignout: false,
        userToken: action.token,
        userType: action.userType,
        userEmail: action.userEmail
      };
    case "SIGN_OUT":
      return {
        ...prevState,
        isSignout: true,
        userToken: null,
        userType: null,
        userEmail: null
      };
    default: {
      throw new Error(`Unhandled auth action type: ${action.type}`);
    }
  }
}
