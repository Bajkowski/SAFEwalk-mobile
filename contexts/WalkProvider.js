import React, { createContext, useReducer, useMemo } from "react";
import { AsyncStorage } from "react-native";

export const WalkContext = createContext();

export function WalkProvider({ children }) {
  const [state, dispatch] = useReducer(walkReducer, {
    // initial states
    isWalkLoading: true,
    isWalkActive: "false", // has to be a string due to AsyncStorage constraints (cannot store boolean)

    // initial walk state
    walkId: null,

    // initial user state
    userEmail: null,
    walkerEmail: null,

    // initial SAFEwalker state
    userSocketId: null,
    walkerSocketId: null,
  });

  // Set up walk ID
  async function setWalkAsActive() {
    try {
      // Store walk ID
      await AsyncStorage.setItem("isWalkActive", "true");

      dispatch({
        type: "SET_WALK_ACTIVE",
        isWalkActive: "true",
      });
    } catch (error) {
      console.error("Error in setIsWalkActive(): " + error);
    }
  }

  // Set up walk ID
  async function setWalkId(walkId) {
    try {
      // Store walk ID
      await AsyncStorage.setItem("walkId", walkId);

      dispatch({
        type: "SET_WALK_ID",
        walkId: walkId,
      });
    } catch (error) {
      console.error("Error in setWalkId(): " + error);
    }
  }

  // Remove walk ID
  async function removeWalkId() {
    try {
      // remove walk ID
      await AsyncStorage.removeItem("WalkId");

      dispatch({ type: "REMOVE_WALK_ID" });
    } catch (error) {
      console.error("Error in removeWalkId(): " + error);
    }
  }

  // Store User information (walkId, email and socket ID)
  async function setUserInfo(walkId, userEmail, userSocketId) {
    try {
      await AsyncStorage.setItem("walkId", walkId);
      await AsyncStorage.setItem("userEmail", userEmail);
      await AsyncStorage.setItem("userSocketId", userSocketId);

      dispatch({
        type: "SET_USER_INFO",
        walkId: walkId,
        userEmail: userEmail,
        userSocketId: userSocketId,
      });
    } catch (error) {
      console.error("Error in seUserInfo(): " + error);
    }
  }

  // Store Safewalker information (email and socket ID)
  async function setWalkerInfo(walkerEmail, walkerSocketId) {
    try {
      await AsyncStorage.setItem("walkerEmail", walkerEmail);
      await AsyncStorage.setItem("walkerSocketId", walkerSocketId);

      dispatch({
        type: "SET_WALKER_INFO",
        walkerEmail: walkerEmail,
        walkerSocketId: walkerSocketId,
      });
    } catch (error) {
      console.error("Error in setWalkerInfo(): " + error);
    }
  }

  // remove all current walk-related information
  async function resetWalkContextState() {
    try {
      await AsyncStorage.removeItem("walkId");
      await AsyncStorage.removeItem("userEmail");
      await AsyncStorage.removeItem("walkerEmail");
      await AsyncStorage.removeItem("userSocketId");
      await AsyncStorage.removeItem("walkerSocketId");

      dispatch({ type: "RESET" });
    } catch (error) {
      console.error("Error in resetContextState(): " + error);
    }
  }

  const walkContextValue = useMemo(() => {
    return {
      // states
      isWalkLoading: state.isWalkLoading,
      isWalkActive: state.isWalkActive,
      walkId: state.walkId,
      userEmail: state.userEmail,
      walkerEmail: state.walkerEmail,
      userSocketId: state.userSocketId,
      walkerSocketId: state.walkerSocketId,

      // functions
      setWalkAsActive,
      setWalkId,
      removeWalkId,
      setUserInfo,
      setWalkerInfo,
      resetWalkContextState,
      dispatch,
    };
  }, [state]);

  return (
    <WalkContext.Provider value={walkContextValue}>
      {children}
    </WalkContext.Provider>
  );
}

function walkReducer(prevState, action) {
  switch (action.type) {
    case "RESTORE_STATE":
      return {
        ...prevState,
        isWalkLoading: false,
        isWalkActive: action.isWalkActive,
        walkId: action.walkId,
        userEmail: action.userEmail,
        walkerEmail: action.walkerEmail,
        userSocketId: action.userSocketId,
        walkerSocketId: action.walkerSocketId,
      };
    case "SET_WALK_ACTIVE":
      return {
        ...prevState,
        isWalkActive: "true",
      };
    case "SET_WALK_ID":
      return {
        ...prevState,
        walkId: action.walkId,
      };
    case "REMOVE_WALK_ID":
      return {
        ...prevState,
        walkId: null,
      };
    case "SET_USER_INFO":
      return {
        ...prevState,
        walkId: action.walkId,
        userEmail: action.userEmail,
        userSocketId: action.userSocketId,
      };
    case "SET_WALKER_INFO":
      return {
        ...prevState,
        walkerEmail: action.walkerEmail,
        walkerSocketId: action.walkerSocketId,
      };
    case "RESET":
      return {
        ...prevState,
        isWalkLoading: false,
        isWalkActive: "false",
        walkId: null,
        userEmail: null,
        userSocketId: null,
        walkerEmail: null,
        walkerSocketId: null,
      };

    default: {
      console.error(`Unhandled walk action type: ${action.type}`);
    }
  }
}
