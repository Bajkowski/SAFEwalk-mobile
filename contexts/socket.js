const io = require("socket.io-client");
module.exports = io("http://127.0.0.1:3000");
// module.exports = io("http://365b17d8.ngrok.io");
