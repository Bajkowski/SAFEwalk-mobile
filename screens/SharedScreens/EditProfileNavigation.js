import React from "react";
import { createStackNavigator } from "@react-navigation/stack";

import Header from "../../components/Header";
import EditProfileScreen from "./EditProfileScreen";

const Stack = createStackNavigator();

export default function EditProfileNavigation() {
  return (
    <Stack.Navigator
      initialRouteName="EditProfile"
      headerMode="screen"
      screenOptions={{
        header: ({ scene, previous, navigation }) => (
          <Header scene={scene} previous={previous} navigation={navigation} />
        ),
      }}
    >
      <Stack.Screen
        name="EditProfile"
        component={EditProfileScreen}
        options={{ title: "Edit Profile" }}
      />
    </Stack.Navigator>
  );
}
