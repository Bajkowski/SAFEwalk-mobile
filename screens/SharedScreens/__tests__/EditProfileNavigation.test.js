import React from "react";
import renderer from "react-test-renderer";
import { render } from "@testing-library/react-native";
import { CardStyleInterpolators } from "@react-navigation/stack";

import { NavigationContainer } from "@react-navigation/native";

import EditProfileNavigation from "../EditProfileNavigation";
import { AuthProvider } from "./../../../contexts/AuthProvider";
import colors from "./../../../constants/colors";

const createTestProps = (props) => ({
    initialRouteName: "EditProfile",
    screenOptions: {
        cardStyleInterpolator: CardStyleInterpolators.forHorizontalIOS,
        headerStyle: {
            backgroundColor: colors.lightorange,
        },
        headerTintColor: "#fff",
        headerTitleAlign: "center",
    },
});

let props;
let component;
let screen;

describe("EditProfileNavigation ", () => {
    beforeEach(() => {
        props = createTestProps();

        component = (
            <AuthProvider>
                <NavigationContainer>
                    <EditProfileNavigation {...props} />;
                </NavigationContainer>
            </AuthProvider>
        );
    });

    it("renders correctly", () => {
        const tree = renderer.create(component).toJSON();
        expect(tree).toMatchSnapshot();
    });
});