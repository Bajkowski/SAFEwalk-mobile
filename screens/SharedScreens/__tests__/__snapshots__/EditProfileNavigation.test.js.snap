// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditProfileNavigation  renders correctly 1`] = `
Array [
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNCSafeAreaView
      onInsetsChange={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        onLayout={[Function]}
        style={
          Object {
            "flex": 1,
            "overflow": "hidden",
          }
        }
      >
        <View
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            accessibilityElementsHidden={false}
            closing={false}
            gestureVelocityImpact={0.3}
            importantForAccessibility="auto"
            onClose={[Function]}
            onGestureBegin={[Function]}
            onGestureCanceled={[Function]}
            onOpen={[Function]}
            onTransitionStart={[Function]}
            pointerEvents="box-none"
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
            transitionSpec={
              Object {
                "close": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 0.01,
                    "restSpeedThreshold": 0.01,
                    "stiffness": 1000,
                  },
                },
                "open": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 0.01,
                    "restSpeedThreshold": 0.01,
                    "stiffness": 1000,
                  },
                },
              }
            }
          >
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <View
                collapsable={false}
                onGestureHandlerEvent={[Function]}
                onGestureHandlerStateChange={[Function]}
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  pointerEvents="none"
                  style={
                    Object {
                      "backgroundColor": "rgb(242, 242, 242)",
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "shadowColor": "#000",
                      "shadowOffset": Object {
                        "height": 1,
                        "width": -1,
                      },
                      "shadowOpacity": 0.3,
                      "shadowRadius": 5,
                      "top": 0,
                      "width": 3,
                    }
                  }
                />
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <TouchableWithoutFeedback
                        accessible={false}
                        onPress={[Function]}
                      >
                        <ScrollView
                          style={
                            Object {
                              "backgroundColor": "#fff",
                              "flex": 1,
                            }
                          }
                        >
                          <SafeAreaView
                            style={
                              Object {
                                "flex": 1,
                                "justifyContent": "flex-end",
                                "marginHorizontal": 75,
                              }
                            }
                          >
                            <View
                              style={
                                Object {
                                  "alignItems": "center",
                                  "marginBottom": 40,
                                  "marginTop": 40,
                                }
                              }
                            >
                              <View
                                accessibilityLabel="NoImageAvatar"
                                style={
                                  Object {
                                    "backgroundColor": "transparent",
                                    "borderRadius": 75,
                                    "height": 150,
                                    "marginBottom": 33.5,
                                    "marginTop": 33.5,
                                    "width": 150,
                                  }
                                }
                                theme={
                                  Object {
                                    "colors": Object {
                                      "disabled": "hsl(208, 8%, 90%)",
                                      "divider": "#bcbbc1",
                                      "error": "#ff190c",
                                      "grey0": "#393e42",
                                      "grey1": "#43484d",
                                      "grey2": "#5e6977",
                                      "grey3": "#86939e",
                                      "grey4": "#bdc6cf",
                                      "grey5": "#e1e8ee",
                                      "greyOutline": "#bbb",
                                      "platform": Object {
                                        "android": Object {
                                          "error": "#f44336",
                                          "primary": "#2196f3",
                                          "secondary": "#9C27B0",
                                          "success": "#4caf50",
                                          "warning": "#ffeb3b",
                                        },
                                        "ios": Object {
                                          "error": "#ff3b30",
                                          "primary": "#007aff",
                                          "secondary": "#5856d6",
                                          "success": "#4cd964",
                                          "warning": "#ffcc00",
                                        },
                                      },
                                      "primary": "#2089dc",
                                      "searchBg": "#303337",
                                      "secondary": "#8F0CE8",
                                      "success": "#52c41a",
                                      "warning": "#faad14",
                                    },
                                  }
                                }
                              >
                                <View
                                  accessibilityIgnoresInvertColors={true}
                                  style={
                                    Object {
                                      "backgroundColor": "#c0c0c0",
                                      "borderRadius": 75,
                                      "flex": 1,
                                      "overflow": "hidden",
                                      "position": "relative",
                                    }
                                  }
                                >
                                  <Image
                                    onLoad={[Function]}
                                    style={
                                      Array [
                                        Object {
                                          "bottom": 0,
                                          "left": 0,
                                          "position": "absolute",
                                          "right": 0,
                                          "top": 0,
                                        },
                                        Object {
                                          "height": null,
                                          "width": null,
                                        },
                                      ]
                                    }
                                    testID="RNE__Image"
                                    theme={
                                      Object {
                                        "colors": Object {
                                          "disabled": "hsl(208, 8%, 90%)",
                                          "divider": "#bcbbc1",
                                          "error": "#ff190c",
                                          "grey0": "#393e42",
                                          "grey1": "#43484d",
                                          "grey2": "#5e6977",
                                          "grey3": "#86939e",
                                          "grey4": "#bdc6cf",
                                          "grey5": "#e1e8ee",
                                          "greyOutline": "#bbb",
                                          "platform": Object {
                                            "android": Object {
                                              "error": "#f44336",
                                              "primary": "#2196f3",
                                              "secondary": "#9C27B0",
                                              "success": "#4caf50",
                                              "warning": "#ffeb3b",
                                            },
                                            "ios": Object {
                                              "error": "#ff3b30",
                                              "primary": "#007aff",
                                              "secondary": "#5856d6",
                                              "success": "#4cd964",
                                              "warning": "#ffcc00",
                                            },
                                          },
                                          "primary": "#2089dc",
                                          "searchBg": "#303337",
                                          "secondary": "#8F0CE8",
                                          "success": "#52c41a",
                                          "warning": "#faad14",
                                        },
                                      }
                                    }
                                  />
                                  <View
                                    accessibilityElementsHidden={false}
                                    importantForAccessibility="yes"
                                    pointerEvents="auto"
                                    style={
                                      Object {
                                        "bottom": 0,
                                        "left": 0,
                                        "opacity": 1,
                                        "position": "absolute",
                                        "right": 0,
                                        "top": 0,
                                      }
                                    }
                                  >
                                    <View
                                      style={
                                        Object {
                                          "alignItems": "center",
                                          "backgroundColor": "transparent",
                                          "flex": 1,
                                          "height": null,
                                          "justifyContent": "center",
                                          "width": null,
                                        }
                                      }
                                      testID="RNE__Image__placeholder"
                                    />
                                  </View>
                                  <View
                                    style={
                                      Object {
                                        "flex": 1,
                                        "height": null,
                                        "width": null,
                                      }
                                    }
                                  />
                                </View>
                              </View>
                            </View>
                            <View
                              accessibilityLabel="Divider 1"
                              style={
                                Object {
                                  "backgroundColor": "black",
                                  "height": 0.5,
                                  "marginBottom": 20,
                                }
                              }
                            />
                            <Text
                              accessibilityLabel="Name"
                              style={
                                Object {
                                  "color": "#7D7D7D",
                                  "fontSize": 29.5,
                                  "marginBottom": 20,
                                  "marginLeft": 20,
                                }
                              }
                            >
                              Name: 
                               
                            </Text>
                            <View
                              accessibilityLabel="Divider 2"
                              style={
                                Object {
                                  "backgroundColor": "black",
                                  "height": 0.5,
                                  "marginBottom": 20,
                                }
                              }
                            />
                            <Text
                              accessibilityLabel="Phone"
                              style={
                                Object {
                                  "color": "#7D7D7D",
                                  "fontSize": 29.5,
                                  "marginBottom": 20,
                                  "marginLeft": 20,
                                }
                              }
                            >
                              Phone Number: 
                              () -
                            </Text>
                            <View
                              accessibilityLabel="Divider 3"
                              style={
                                Object {
                                  "backgroundColor": "black",
                                  "height": 0.5,
                                  "marginBottom": 20,
                                }
                              }
                            />
                            <Text
                              accessibilityLabel="Interests"
                              style={
                                Object {
                                  "color": "#7D7D7D",
                                  "fontSize": 29.5,
                                  "marginBottom": 20,
                                  "marginLeft": 20,
                                }
                              }
                            >
                              Interests: 
                              
                            </Text>
                            <View
                              accessibilityLabel="Divider 4"
                              style={
                                Object {
                                  "backgroundColor": "black",
                                  "height": 0.5,
                                  "marginBottom": 20,
                                }
                              }
                            />
                            <View
                              style={
                                Object {
                                  "height": 267,
                                  "justifyContent": "space-around",
                                }
                              }
                            >
                              <View
                                style={
                                  Object {
                                    "borderRadius": 3,
                                  }
                                }
                              >
                                <TouchableOpacity
                                  accessibilityLabel="EditButton"
                                  activeOpacity={0.3}
                                  disabled={false}
                                  onPress={[Function]}
                                >
                                  <View
                                    style={
                                      Object {
                                        "alignItems": "center",
                                        "backgroundColor": "#fd6e11",
                                        "borderColor": "#2089dc",
                                        "borderRadius": 5,
                                        "borderWidth": 0,
                                        "flexDirection": "row",
                                        "height": 80,
                                        "justifyContent": "center",
                                        "padding": 8,
                                      }
                                    }
                                  >
                                    <Text
                                      style={
                                        Object {
                                          "color": "white",
                                          "fontSize": 26.5,
                                          "paddingBottom": 1,
                                          "paddingTop": 2,
                                          "textAlign": "center",
                                        }
                                      }
                                    >
                                      Edit
                                    </Text>
                                  </View>
                                </TouchableOpacity>
                              </View>
                            </View>
                          </SafeAreaView>
                        </ScrollView>
                      </TouchableWithoutFeedback>
                    </View>
                    <View
                      pointerEvents="box-none"
                    >
                      <View
                        accessibilityElementsHidden={false}
                        importantForAccessibility="auto"
                        onLayout={[Function]}
                        pointerEvents="box-none"
                        style={null}
                      >
                        <SafeAreaView
                          style={
                            Array [
                              Object {
                                "backgroundColor": "#ed7c3a",
                                "elevation": 4,
                                "zIndex": 0,
                              },
                              Object {
                                "shadowColor": "#000000",
                                "shadowOffset": Object {
                                  "height": 3,
                                  "width": 0,
                                },
                                "shadowOpacity": 0.24,
                                "shadowRadius": 4,
                              },
                            ]
                          }
                        >
                          <View
                            style={
                              Object {
                                "alignItems": "center",
                                "backgroundColor": "#ed7c3a",
                                "elevation": 0,
                                "flexDirection": "row",
                                "height": 56,
                                "marginTop": 0,
                                "paddingHorizontal": 4,
                              }
                            }
                          >
                            <TouchableHighlight
                              accessibilityComponentType="button"
                              accessibilityRole="button"
                              accessibilityStates={Array []}
                              accessibilityTraits="button"
                              activeOpacity={0.85}
                              centered={true}
                              delayPressOut={100}
                              disabled={false}
                              hitSlop={
                                Object {
                                  "bottom": 6,
                                  "left": 6,
                                  "right": 6,
                                  "top": 6,
                                }
                              }
                              onPress={[Function]}
                              style={
                                Array [
                                  Object {
                                    "overflow": "hidden",
                                  },
                                  Array [
                                    Object {
                                      "alignItems": "center",
                                      "justifyContent": "center",
                                      "margin": 6,
                                      "overflow": "hidden",
                                    },
                                    Object {
                                      "borderRadius": 18,
                                      "height": 36,
                                      "width": 36,
                                    },
                                    undefined,
                                    undefined,
                                  ],
                                ]
                              }
                              underlayColor="rgba(255, 255, 255, 0.16)"
                            >
                              <View>
                                <View
                                  style={
                                    Array [
                                      Object {
                                        "alignItems": "center",
                                        "justifyContent": "center",
                                      },
                                      Object {
                                        "height": 24,
                                        "width": 24,
                                      },
                                    ]
                                  }
                                >
                                  <View
                                    style={
                                      Object {
                                        "bottom": 0,
                                        "left": 0,
                                        "opacity": 1,
                                        "position": "absolute",
                                        "right": 0,
                                        "top": 0,
                                        "transform": Array [
                                          Object {
                                            "rotate": "0deg",
                                          },
                                        ],
                                      }
                                    }
                                  >
                                    <Text />
                                  </View>
                                </View>
                              </View>
                            </TouchableHighlight>
                            <TouchableWithoutFeedback
                              disabled={true}
                            >
                              <View
                                style={
                                  Array [
                                    Object {
                                      "flex": 1,
                                      "paddingHorizontal": 12,
                                    },
                                    Array [
                                      Object {
                                        "marginLeft": 8,
                                      },
                                      Object {
                                        "alignItems": "center",
                                      },
                                      undefined,
                                    ],
                                  ]
                                }
                              >
                                <Text
                                  accessibilityRole="header"
                                  accessibilityTraits="header"
                                  accessible={true}
                                  numberOfLines={1}
                                  style={
                                    Array [
                                      Object {
                                        "color": "#000000",
                                        "fontFamily": "System",
                                        "fontWeight": "400",
                                        "textAlign": "left",
                                      },
                                      Array [
                                        Object {
                                          "color": "#000000",
                                          "fontFamily": "System",
                                          "fontWeight": "400",
                                        },
                                        Object {
                                          "fontSize": 17,
                                        },
                                        Object {
                                          "alignSelf": "center",
                                          "color": "#fff",
                                        },
                                      ],
                                    ]
                                  }
                                >
                                  Edit Profile
                                </Text>
                              </View>
                            </TouchableWithoutFeedback>
                            <View
                              style={
                                Object {
                                  "width": 48,
                                }
                              }
                            />
                          </View>
                        </SafeAreaView>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RNCSafeAreaView>
  </View>,
  ";",
]
`;
